### ISSUE TITLE
Unable to view boxes in Sharktopoda

### ISSUE THREAD

**Issue by KSchlining** [mbari-org/vars-annotation#174]
When we are cleaning up the ML proposals often when we finish with one file and then open our next file the boxes don't show up in Sharktopoda. We have to restart VARS 2 or 3 (sometimes more) times before the boxes show up again.

**Comment by hohonuuli** [comment]
This dive file was not showing boxes:
V4394_20220427T172554Z_prores.mov

**Comment by hohonuuli** [comment]
This bug does not appear when opening the same video repeatedly. It occurs when you open a second/different video in VARS>

**Comment by KSchlining** [comment]
Yes, it often (not always) does occur repeatedly. Sometimes you need to restart 2 or 3 times. This issue has been noticed by Kyra, Kris, & Lonny.

**Comment by lonnylundsten** [comment]
I'm finding that you can click the 'refresh data' button -- might take 1 or 2 times -- and eventually the localizations will show. This is a better option than having to restart vars and s2 completely.

**Comment by KSchlining** [comment]
This works to make the localizations show up, but all my buttons in my concept map disappear. So, I've been restarting VARS for pretty much every file.

**Comment by hohonuuli** [comment]
From @kwalz:

> Latest log, I just opened a file (V4444 20221028T203700Z, prores file) and no boxes drawn from VARS anno to Sharktopoda.

[vars-annotation.20240403.log](https://github.com/mbari-org/vars-annotation/files/14860073/vars-annotation.20240403.log)

**Comment by hohonuuli** [comment]
From @kwalz 

> A copy of my VARS logs--this morning it has been particularly difficult to open VARS, either the boxes are not being drawn in sharktopoda or I get the boxes but my buttons are missing, has been taking me 4-5 times to restart VARS anno to get both of these items at the same time. Unfortunately, for me I don't get my buttons to show up just by resizing the VARS anno window.

[vars-annotation.20240409.log](https://github.com/mbari-org/vars-annotation/files/14924678/vars-annotation.20240409.log)

**Comment by kwalz** [comment]
@NancyJS @hohonuuli I'm back to editing proposals in VARS Annotation and this issue (above) is a show stopper, I have reopen VARS Anno and the file numerous time to get the boxes to show up (after I've reopened VARS annotation a few times to get my maps populated). Can we get these two issues fixes as soon as possible?
kris

**Comment by lonnylundsten** [comment]
This is still happening as of Jun 24, 2024. I refreshed VARS 15 times this morning and boxes still did not show in S2. I will now close and re-open VARS, hopefully they show. Unfortunately many hours of the day are burned dealing with this issue. Constantly having to restart is not an option when we're trying to be productive.

**Comment by lonnylundsten** [comment]
Restarted VARS, no boxes. Restarted VARS and S2, still no boxes.

**Comment by kwalz** [comment]
@hohonuuli  A note on this issue, Im having where some of my boxes are drawing to shark-2 but there are some that are not, mixed throughout the video file. A restart fixes this but just a note that we need to thoroughly test this since it may seem like they are drawing on shark-2 but not all are drawing for the file.

**Comment by kwalz** [comment]
Boxes are drawn for the first file that I open with VARS anno but when I go to open another video file, boxes are not on the second file that I open or any other files until I restart VARS anno (consistently drawn for first file).

**Comment by hohonuuli** [comment]
Need the video lab to do some more testing with v2.1.2

**Comment by kwalz** [comment]
@hohonuuli This is still happening for us in lab, @rissalemon and I had this issue again today, restarted VARS anno and sharktopoda to get boxes to show up.

**Comment by hohonuuli** [comment]
@kwalz Thanks for letting me know. Hope to dig into some of these box issues after the marine imaging workshop.

**Comment by rissalemon** [comment]
This is still a big issue. I am having to restart multiple (5 to 6) times a day. It is 2pm right now and I have had to refresh at least three times in between files, usually resulting in having to restart so I get both my concepts and the localization boxes. I think it wouldn't be as big of an issue with normal annotation, but I am cleaning up the rest of the Paull dives for the EXPRESS group  so these need to be cleaned up to match previously annotated dives. The new aspect is that upon the second or third refresh, I get the beach ball loading icon until I have to force quit.

**Comment by hohonuuli** [comment]
@rissalemon Can you give the URL to one of the files that you are experiencing this issue with? I have a suspicion there might be bounding boxes with zero-width/height or out of the video bounds that may be contributing to this issue. But I need to have a pointer to a video where this is occurring to confirm this.

**Comment by rissalemon** [comment]
@hohonuuli This is one that I was working on yesterday, but has been going on for a while.  The force quite aspect is new though...

http://m3.shore.mbari.org/videos/M3/mezzanine/DocRicketts/2021/08/1377/D1377_20210815T155838Z_h264.mp4

**Comment by kwalz** [comment]
Probably not a Mac chip vs intel issue, boxes were not drawing for me on my new laptop which is m2 and also weren't drawing on my desktop machine which is intel.

**Comment by kwalz** [comment]
@hohonuuli This issue is still happening, wanted to note here.  It happens on both my mac studio and my laptop.

**Comment by dingosky** [comment]
This issue is addressed in Sharktopoda 2.2.2 and proposed changes to VARS Annotation. Stay tuned for feedback from @hohonuuli
